### PR TITLE
Update to apollo v3

### PR DESCRIPTION
**Pull request by Sartonon**
## Description :sparkles:

- Update apollo to v3 + related packages

## Issues :bug:
**[TH-1142](https://helsinkisolutionoffice.atlassian.net/browse/TH-1142):** 

## Additional notes :spiral_notepad:

### CODE REVIEW

File: src/domain/app/BrowserApp.tsx [hunk 1]
```diff
@@ -24,11 +24,11 @@ const BrowserApp: React.FC = () => {
   return (
     <BrowserRouter>
       <ApolloProvider client={apolloClient}>
-        <ApolloHooksProvider client={apolloClient}>
-          <MatomoProvider value={instance}>
-            <App />
-          </MatomoProvider>
-        </ApolloHooksProvider>
+        {/* <ApolloHooksProvider client={apolloClient}> */}
```
Comment (nikomakela) [inline src/domain/app/BrowserApp.tsx:27]: Is there a reason why these are commented out?

File: src/domain/app/apollo/apolloClient.ts [hunk 2]
```diff
@@ -1,22 +1,42 @@
...
+        image(_, { args, toReference }) {
+          return toReference({
+            __typename: 'Place',
+            id: args?.id,
+          });
+        },
       },
     },
   },
+  // cacheRedirects: {
```
Comment (nikomakela) [inline src/domain/app/apollo/apolloClient.ts:30]: Again some commented code

File: src/domain/app/BrowserApp.tsx [hunk 3]
```diff
@@ -2,10 +2,10 @@ import '../../common/translation/i18n/init.client';
 import '../../globals';
 import 'react-toastify/dist/ReactToastify.css';
 
+import { ApolloProvider } from '@apollo/client';
+// import { ApolloProvider as ApolloHooksProvider } from '@apollo/react-hooks';
```
Comment (nikomakela) [inline src/domain/app/BrowserApp.tsx:6]: commented code

File: src/domain/app/apollo/apolloClient.ts [hunk 4]
```diff
@@ -1,22 +1,42 @@
...
+  typePolicies: {
     Query: {
-      keywordDetails: (_, args, { getCacheKey }) => {
-        return getCacheKey({ __typename: 'Keyword', id: args.id });
-      },
-      placeDetails: (_, args, { getCacheKey }) => {
-        return getCacheKey({ __typename: 'Place', id: args.id });
+      fields: {
+        event(_, { args, toReference }) {
+          return toReference({
```
Comment (nikomakela) [inline src/domain/app/apollo/apolloClient.ts:16]: if the cache is replaced, isn't the mergeIntoStore: true needed?

File: src/domain/app/BrowserApp.tsx (hunk 1 above)
Comment (Sartonon) [reply src/domain/app/BrowserApp.tsx]: Deleted, forgot to push

File: src/domain/app/apollo/apolloClient.ts (hunk 2 above)
Comment (Sartonon) [reply src/domain/app/apollo/apolloClient.ts]: Deleted, forgot to push

File: src/domain/app/apollo/apolloClient.ts (hunk 4 above)
Comment (Sartonon) [reply src/domain/app/apollo/apolloClient.ts:16]: Actually we are already fetching from cache here, not sure if adding that single keyword to cache is needed.

**Review by nikomakela — APPROVED** [review]

**Pull request merged**